packages/core/src/external/ehr/command/get-appointments/ehr-get-appointments-direct.ts

```diff
@@ -0,0 +1,35 @@
+import { BadRequestError } from "@metriport/shared";
+import { getAppointments as getAppointmentsCanvas } from "../../canvas/command/get-appointments";
+import {
+  Appointment,
+  AppointmentMethods,
+  EhrGetAppointmentsHandler,
+  GetAppointmentsClientRequest,
+  GetAppointmentsRequest,
+} from "./ehr-get-appointments";
+
+export class EhrGetAppointmentsDirect implements EhrGetAppointmentsHandler {
+  async getAppointments<T extends Appointment>({
+    method,
+    ...params
+  }: GetAppointmentsRequest): Promise<T[]> {
+    const handler = getEhrGetAppointmentsHandler(method);
+    return (await handler({ ...params })) as T[];
+  }
+}
+
+type GetAppointments = (params: GetAppointmentsClientRequest) => Promise<Appointment[]>;
+
+type AppointmentMethodsMap = Record<AppointmentMethods, GetAppointments | undefined>;
+
+const ehrGetAppointmentsMap: AppointmentMethodsMap = {
+  [AppointmentMethods.canvasGetAppointments]: getAppointmentsCanvas,
+};
```

Comment:
This is not ideal, as we're adding EHR-specific information in the global EHR code.

Not a blocker for now, but I'd create an issue to map this and add a TODO here.

Example of solution: could have a "setup" step/function so that each EHR implementation is required to pass certain functions to that code, and that effectively initializes the EHR global code so it can direct the execution flow to said functions based on the context/parameters of the execution.

Reply:
> if there's a setup step for each EHR for example, then it feels like we're just moving the map up the chain but not removing it.

This is better IMO. The fact the global code has to know about each EHR is not ideal, all new EHRs will require changes on these global files and the global code gets mingled w/ EHR-specific.

Not a huge deal if we can keep these enforced at compilation time and there's a fair separation of concerns/clean code.

Thanks for creating the ticket.